### ISSUE TITLE
Ability to sign in with something other than email address

### ISSUE THREAD

**Issue by YUzhva** [neroniaky/angular2-token#157]
Hey hi.

According to the documentation
https://github.com/neroniaky/angular2-token#signin
For .signIn() method, email and password are required and userType - optional.

I set up my api to accept **mobile_number** instead of email for user authentication:
https://github.com/plataformatec/devise/wiki/How-To:-Allow-users-to-sign-in-using-their-username-or-email-address

Is there a way to use '**mobile_number**' (or another custom key) to sign in with something other than email address?

**Comment by neroniaky** [comment]
Hey @YUzhva, `angular2-token` doesn't check if email is actually an email, which you could use for some kind a temporary hack (Don't know if Device supports this). Changing the key won't work, because `.signIn()` expects the `SignInData`-Type.

```
export interface SignInData {
    email:                  string;
    password:               string;
    userType?:              string;
}
```

Gonna put this on my ToDo-List.

**Comment by SampadaBhagat** [comment]
Any updates on this? I need to send an object in the signIn call.

**Comment by brunoluigi** [comment]
Folks, started dealing with this issue in the PR #402 but I need some help with some design issues, mainly how to treat this custom field in the other flows such as register, password recovery, etc

**Comment by neroniaky** [comment]
Fixed in 6.0.
